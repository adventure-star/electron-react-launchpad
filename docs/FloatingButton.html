<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>JSDoc: Class: FloatingButton</title>

    <script src="scripts/prettify/prettify.js"> </script>
    <script src="scripts/prettify/lang-css.js"> </script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link type="text/css" rel="stylesheet" href="styles/prettify-tomorrow.css">
    <link type="text/css" rel="stylesheet" href="styles/jsdoc-default.css">
</head>

<body>

<div id="main">

    <h1 class="page-title">Class: FloatingButton</h1>

    




<section>

<header>
    
        <h2>FloatingButton</h2>
        
    
</header>

<article>
    <div class="container-overview">
    
        

    

    <h4 class="name" id="FloatingButton"><span class="type-signature"></span>new FloatingButton<span class="signature">(props)</span><span class="type-signature"></span></h4>

    



<div class="description">
    The FloatingButton component (MaterializeCSS project)
</div>









    <h5>Parameters:</h5>
    

<table class="params">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>props</code></td>
            

            <td class="type">
            
                
<span class="param-type">*</span>


            
            </td>

            

            

            <td class="description last">React property object</td>
        </tr>

    
    </tbody>
</table>






    <h5 class="subsection-title">Properties:</h5>

    

<table class="props">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>className</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">Additional classes for the component</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>animate</code></td>
            

            <td class="type">
            
                
<span class="param-type">boolean</span>


            
            </td>

            

            

            <td class="description last">Whether or not a hover will trigger an animation</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>onClick</code></td>
            

            <td class="type">
            
                
<span class="param-type">function</span>


            
            </td>

            

            

            <td class="description last">The onclick event</td>
        </tr>

    
    </tbody>
</table>




<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="renderer_components_button_floatingbutton.jsx.html">renderer/components/button/floatingbutton.jsx</a>, <a href="renderer_components_button_floatingbutton.jsx.html#line17">line 17</a>
    </li></ul></dd>
    

    

    

    
</dl>















    <h5>Example</h5>
    
    <pre class="prettyprint"><code>An example can be found here: http://materializecss.com/buttons.html
         import FloatingButton from './components/floatingbutton'
         &lt;FloatingButton>Button text&lt;/FloatingButton></code></pre>



    
    </div>

    

    

    

     

    

    

    

    

    
</article>

</section>




</div>

<nav>
    <h2><a href="index.html">Home</a></h2><h3>Classes</h3><ul><li><a href="BaseTextInput.html">BaseTextInput</a></li><li><a href="Card.html">Card</a></li><li><a href="CardAction.html">CardAction</a></li><li><a href="CardContent.html">CardContent</a></li><li><a href="CardImage.html">CardImage</a></li><li><a href="CardPanel.html">CardPanel</a></li><li><a href="CardStacked.html">CardStacked</a></li><li><a href="CardTitle.html">CardTitle</a></li><li><a href="Collection.html">Collection</a></li><li><a href="CollectionHeader.html">CollectionHeader</a></li><li><a href="CollectionItem.html">CollectionItem</a></li><li><a href="CollectionLink.html">CollectionLink</a></li><li><a href="EmailInput.html">EmailInput</a></li><li><a href="FixedButton.html">FixedButton</a></li><li><a href="FloatingButton.html">FloatingButton</a></li><li><a href="PasswordInput.html">PasswordInput</a></li><li><a href="RaisedButton.html">RaisedButton</a></li><li><a href="TextInput.html">TextInput</a></li><li><a href="WindowManager.html">WindowManager</a></li></ul><h3>Global</h3><ul><li><a href="global.html#getWindowManager">getWindowManager</a></li><li><a href="global.html#loadFile">loadFile</a></li><li><a href="global.html#openFile">openFile</a></li><li><a href="global.html#showOpenDialog">showOpenDialog</a></li></ul>
</nav>

<br class="clear">

<footer>
    Documentation generated by <a href="https://github.com/jsdoc3/jsdoc">JSDoc 3.4.0</a> on Tue Jul 18 2017 21:53:28 GMT+0200 (CEST)
</footer>

<script> prettyPrint(); </script>
<script src="scripts/linenumber.js"> </script>
</body>
</html>