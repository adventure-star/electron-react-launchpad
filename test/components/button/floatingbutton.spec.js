import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import FloatingButton from '../../../src/renderer/components/button/floatingbutton'

function testFunction() {
    // DO NOTHING

    return "";
}

function setupStandard() {
    const renderer = ReactTestRenderer.create(<FloatingButton onClick={testFunction} className='testClass'>Test button text</FloatingButton>);
    const component = renderer.toJSON();

    return {
        component
    };
}

describe('Button - FloatingButton component', () => {
    it('should be of type button', () => {
        const { component } = setupStandard();
        expect(component.type).toBe('button');
    });

    it('should have only the classes "btn-floating" and "testClass"', () => {
        const { component } = setupStandard();
        expect(component.props.className).toBe('btn-floating testClass');
    });

    it('should have an onClick function', () => {
        const { component } = setupStandard();
        expect(component.props.onClick).toBe(testFunction);
    });

    it('children should be the text "Test content"', () => {
        const { component } = setupStandard();
        expect(component.children[0]).toBe('Test button text');
    });

    it('Matches/Renders a snapshot', () => {
        const { component } = setupStandard();
        expect(component).toMatchSnapshot();
    });
});
