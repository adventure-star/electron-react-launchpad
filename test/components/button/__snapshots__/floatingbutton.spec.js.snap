// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button - FloatingButton component Matches/Renders a snapshot 1`] = `
<button
  className="btn-floating testClass"
  onClick={[Function]}
>
  Test button text
</button>
`;
